packages/block-editor/src/components/inserter/search-results.js

```diff
@@ -89,10 +121,16 @@ function InserterSearchResults( {
 		if ( maxBlockTypesToShow === 0 ) {
 			return [];
 		}
+
 		let orderedItems = orderBy( blockTypes, 'frecency', 'desc' );
-		if ( ! filterValue && orderInitialBlockItems ) {
-			orderedItems = orderInitialBlockItems( orderedItems );
+
+		if ( ! filterValue && parentInserterPriority ) {
```

Comment:
This functionality would be great to add in slash inserter too [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/autocompleters/block.js#L67).

Reply:
This would involve extracting a hook for the retrieval of `prioritisedInserterBlocks` and also moving the sorting function to a shared utils location.

Once done this should be simple enough. I can tackle tomorrow.

Reply:
This would involve extracting a hook for the retrieval of `prioritisedInserterBlocks` and also moving the sorting function to a shared utils location.

Once done this should be simple enough. I can tackle tomorrow.

Reply:
> This functionality would be great to add in slash inserter too [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/autocompleters/block.js#L67).

Let's do this in a followup.

Reply:
We can, but in general this is a tiny change. I tried it before commenting. We just need to find a good place for the sorting function(ex utils or something). Also we don't even need a new hook, as is just a `select` call.

Reply:
Created an issue here: https://github.com/WordPress/gutenberg/issues/50635